### PR TITLE
Refactor Parser

### PR DESCRIPTION
**Pull request by lackmoo**
Refactor Parser to adhere to the good code quality standard.

### CODE REVIEW

File: src/main/java/spinbox/Parser.java [hunk 1]
```diff
@@ -43,98 +45,164 @@ public static void setPageTrace(ArrayDeque<String> pageTraceNew) {
...
+     */
+    private static String emptyPageComponentAppender(String pageData, ArrayDeque<String> tempPageTrace,
+                                                     String lastElement) {
+        String fullPageData;
+        fullPageData = pageData.concat(lastElement);
+        tempPageTrace.removeLast();
+        if (tempPageTrace.size() != 0) {
+            String lastComponent = tempPageTrace.getLast();
+            String finalFullPageData;
+            finalFullPageData = fullPageData.concat(" " + lastComponent.toUpperCase());
```
Comment (rssujay) [inline src/main/java/spinbox/Parser.java]: Might want to declare + initialize in one line

File: src/main/java/spinbox/Parser.java [hunk 2]
```diff
@@ -43,98 +45,164 @@ public static void setPageTrace(ArrayDeque<String> pageTraceNew) {
...
+            LOGGER.info("Exiting static method emptyPageComponentAppender with pageData: " + finalFullPageData);
```
Comment (rssujay) [inline src/main/java/spinbox/Parser.java]: Pull out magic string

File: src/main/java/spinbox/Parser.java [hunk 3]
```diff
@@ -43,98 +45,164 @@ public static void setPageTrace(ArrayDeque<String> pageTraceNew) {
...
+            fullPageData = pageData.concat(firstPageComponent);
+            break;
+        case "modules":
+            fullPageData = emptyPageComponentAppender(pageData, tempPageTrace, firstPageComponent);
+            break;
+        default:
+            // pageData is a module code
+            fullPageData = pageData.concat("modules " + firstPageComponent.toUpperCase());
+        }
+        LOGGER.info("Exiting static method singlePageComponentAppender with pageData: " + fullPageData);
```
Comment (rssujay) [inline src/main/java/spinbox/Parser.java]: LOGGER.entering / LOGGER.exiting seems more appropriate here, unless these are more important to log

File: src/main/java/spinbox/Parser.java [hunk 4]
```diff
@@ -43,98 +45,164 @@ public static void setPageTrace(ArrayDeque<String> pageTraceNew) {
...
-                pageData = pageData.concat("modules " + pageComponent[1].toUpperCase());
-            } else {
-                throw new InputException(INVALID_COMMAND);
-            }
+            pageData = singlePageComponentAppender(pageData, firstPageComponent, tempPageTrace);
+        } else if (pageComponent.length == 2) {
+            String secondPageComponent = pageComponent[1];
+            pageData = fullPageComponentAppender(pageData, firstPageComponent, secondPageComponent);
+        } else {
+            LOGGER.warning("Invalid command entered, error propagated upwards.");
```
Comment (rssujay) [inline src/main/java/spinbox/Parser.java]: Magic string take out as constant

File: src/main/java/spinbox/Parser.java (hunk 1 above)
Comment (lackmoo) [reply src/main/java/spinbox/Parser.java]: Okay I have fixed this.

File: src/main/java/spinbox/Parser.java (hunk 2 above)
Comment (lackmoo) [reply src/main/java/spinbox/Parser.java]: Magic String created.

File: src/main/java/spinbox/Parser.java (hunk 3 above)
Comment (lackmoo) [reply src/main/java/spinbox/Parser.java]: Changed to the above methods.

File: src/main/java/spinbox/Parser.java (hunk 4 above)
Comment (lackmoo) [reply src/main/java/spinbox/Parser.java]: Magic String created.

**Pull request merged**